tests/unit/test_collector.py

```diff
@@ -355,14 +357,63 @@ def test_parse_links__yanked_reason(anchor_html, expected):
     page = HTMLPage(
         html_bytes,
         encoding=None,
-        url='https://example.com/simple/',
+        # parse_links() is cached by url, so we inject a random uuid to ensure
+        # the page content isn't cached.
+        url='https://example.com/simple-{}/'.format(uuid.uuid4()),
     )
     links = list(parse_links(page))
     link, = links
     actual = link.yanked_reason
     assert actual == expected
 
 
+@skip_if_python2
+def test_parse_links_caches_same_page_by_url():
+    html = (
+        # Mark this as a unicode string for Python 2 since anchor_html
```

Comment:
This comment (and the `u''` prefix) aren't relevant in a `@skip_if_python2` test ^^

Reply:
Thank you!!! Will fix!